src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/scanner/StreamReader.kt

```diff
@@ -98,7 +98,7 @@ class StreamReader(
             index = index,
             line = line,
             column = column,
-            codepoints = codePointsWindow.toList(),
+            codepoints = codePointsWindow.toReadOnlyListAdapter(),
```

Comment:
There's also [`asList()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/as-list.html). I've tried it and I think it works the same. WDYT?

Reply:
Yea, indeed. Completely forgot about that

Reply:
Interesting note though:

For the JS target the method `asList` returns a mutable list

```kotlin
public actual inline fun IntArray.asList(): List<Int> {
    return this.unsafeCast<Array<Int>>().asList()
}

public actual fun <T> Array<out T>.asList(): List<T> {
    return ArrayList<T>(this.unsafeCast<Array<Any?>>())
}
```